transports/webrtc/src/upgrade.rs

```diff
@@ -59,17 +59,23 @@ pub async fn outbound(
     log::debug!("created SDP offer for outbound connection: {:?}", offer.sdp);
     peer_connection.set_local_description(offer).await?;
 
-    let answer = sdp::answer(addr, &remote_fingerprint);
+    let answer = sdp::answer(addr, &server_fingerprint);
     log::debug!(
         "calculated SDP answer for outbound connection: {:?}",
         answer
     );
     peer_connection.set_remote_description(answer).await?; // This will start the gathering of ICE candidates.
 
+    // Note the roles are reversed because it allows the server (webrtc connection responder) to
+    // send application data 0.5 RTT earlier.
     let data_channel = create_substream_for_noise_handshake(&peer_connection).await?;
-
-    let peer_id =
-        noise::outbound(id_keys, data_channel, our_fingerprint, remote_fingerprint).await?;
+    let peer_id = noise::inbound(
```

Comment:
If we are reversing the roles here then I think we should still call this `outbound` but reverse it inside. From this protocols perspective, this is still the outbound noise handshake.